src/app/Library/CrudPanel/Traits/Tabs.php

```diff
@@ -106,56 +106,77 @@ public function isLastTab($label)
     }
 
     /**
+     * @deprecated Do not use this method as it will be removed in future versions!
+     * Instead, use $this->getElementsWithoutATab($this->getCurrentFields())
+     *
      * @return \Illuminate\Support\Collection
      */
     public function getFieldsWithoutATab()
     {
-        $all_fields = $this->getCurrentFields();
+        return $this->getElementsWithoutATab($this->getCurrentFields());
+    }
 
-        $fields_without_a_tab = collect($all_fields)->filter(function ($value, $key) {
+    /**
+     * @return \Illuminate\Support\Collection
+     */
+    public function getElementsWithoutATab(array $elements)
+    {
+        return collect($elements)->filter(function ($value) {
             return ! isset($value['tab']);
         });
+    }
 
-        return $fields_without_a_tab;
+    /**
+     * @return array|\Illuminate\Support\Collection
+     */
+    public function getTabFields(string $label)
+    {
+        return $this->getTabElements($label, $this->getCurrentFields());
     }
 
     /**
-     * @param $label
      * @return array|\Illuminate\Support\Collection
      */
-    public function getTabFields($label)
+    public function getTabColumns(string $label)
     {
-        if ($this->tabExists($label)) {
-            $all_fields = $this->getCurrentFields();
+        return $this->getTabElements($label, $this->columns());
+    }
 
-            $fields_for_current_tab = collect($all_fields)->filter(function ($value, $key) use ($label) {
+    /**
+     * @return array|\Illuminate\Support\Collection
+     */
+    public function getTabElements(string $label, array $elements)
+    {
+        if ($this->tabExists($label)) {
+            return collect($elements)->filter(function ($value) use ($label) {
                 return isset($value['tab']) && $value['tab'] == $label;
             });
-
-            return $fields_for_current_tab;
         }
 
         return [];
     }
 
-    /**
-     * @return array
-     */
-    public function getTabs()
+    public function getTabs(): array
     {
         $tabs = [];
-        $fields = $this->getCurrentFields();
 
-        $fields_with_tabs = collect($fields)
-            ->filter(function ($value, $key) {
+        collect($this->getCurrentFieldsOrColumns())
+            ->filter(function ($value) {
                 return isset($value['tab']);
             })
-            ->each(function ($value, $key) use (&$tabs) {
+            ->each(function ($value) use (&$tabs) {
                 if (! in_array($value['tab'], $tabs)) {
                     $tabs[] = $value['tab'];
                 }
             });
 
         return $tabs;
     }
+
+    public function getCurrentFieldsOrColumns(): array
+    {
+        return $this->getCurrentOperation() === 'show'
```

Comment:
🚩 what Pedro said - this should be done differently. We can even get BOTH columns and fields, I think... right? So that it's not tied to a particular operation.

Reply:
Which one you get first? 

If developer setup fields and columns inside `setup()` without specifying any operations this will make a mess for sure if we get both. 

OR

we save them already keyed ?
```php
$tabbed => [
'fields' => [ .... ]
'columns' => [ ... ]
];
```